hardware/diskinfo.py

```diff
@@ -43,42 +43,39 @@ def get_disk_info(name, sizes, hw_lst):
                  'queue/optimal_io_size', 'queue/physical_block_size',
                  'queue/rotational', 'queue/nr_requests']
     for info in info_list:
-        disk_sys_info = '/sys/block/%s/%s' % (name, info)
+        disk_sys_info = f'/sys/block/{name}/{info}'
         # revision can be explicitly named
         if not os.path.exists(disk_sys_info) and info == 'device/rev':
             info = 'device/revision'
         # for nvme devices we can have a nested device dir
         if not os.path.exists(disk_sys_info):
-            disk_sys_info = '/sys/block/%s/device/%s' % (name, info)
+            disk_sys_info = f'/sys/block/{name}/device/{info}'
         try:
             with open(disk_sys_info, 'r') as dev:
                 hw_lst.append(('disk', name, info.split('/')[1],
                                dev.readline().rstrip('\n').strip()))
         except Exception as exc:
-            sys.stderr.write(
-                'Failed retrieving disk information %s for %s: %s\n' % (
-                    info, name, str(exc)))
+            sys.stderr.write(f"Failed retrieving disk information {info} "
+                             f"for {name}: {str(exc)}\n")
```

Comment:
Please don't use str(). It's completely redundant, and I'm not sure if it actually works in formatted strings.